### PR TITLE
[40595] Create a Multishop management page

### PR DESCRIPTION
**Pull request by rleenders**
# Story Reference

[40595](https://app.clubhouse.io/ns8/story/40595)

## Summary

add activate/deactivate buttons to the existing magento store management page

Screen Shot 2020-08-17 at 1 25 48 PM

## Author Checklist

I have added/updated:

- [ ] Tests
- [ ] Documentation
- [ ] Release notes (title of this PR)

## Version Bump

- [ ] Patch (bug fix - backwards compatible)
- [ ] Minor (new functionality - backwards compatible)
- [ ] Major (significant change - not backwards compatible)

### CODE REVIEW

File: view/adminhtml/layout/adminhtml_system_store_grid_block.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                    <argument name="save_parameters_in_session" xsi:type="string">1</argument>
+                    <argument name="dataSource" xsi:type="object" shared="false">Magento\Store\Model\ResourceModel\Website\Grid\Collection</argument>
+                </arguments>
+                <referenceBlock class="Magento\Backend\Block\Widget\Grid\ColumnSet" name="adminhtml.system.store.grid.columnSet" as="grid.columnSet">
+                    <arguments>
+                        <argument name="id" xsi:type="string">storeGrid</argument>
+                    </arguments>
+                    <block class="Magento\Backend\Block\Widget\Grid\Column" name="adminhtml.system.store.grid.columnSet.protect_active" as="protect_active">
+                        <arguments>
+                            <argument name="header" xsi:type="string" translate="true">Protect Status</argument>
```
Comment (rob-ns8) [inline view/adminhtml/layout/adminhtml_system_store_grid_block.xml:23]: Should this maybe say `NS8 Protect Status`?

Comment (rleenders) [reply view/adminhtml/layout/adminhtml_system_store_grid_block.xml:23]: sure

File: Controller/Adminhtml/Shop/Manage.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+     */
+    public function execute(): string
+    {
+        $body = $this->request->getPostValue();
+        if (isset($body['activate'])) {
+            $this->setup->activateShop($body['activate']);
+        }
+        if (isset($body['deactivate'])) {
+            $this->setup->deactivateShop($body['deactivate']);
+        }
```
Comment (rob-ns8) [inline Controller/Adminhtml/Shop/Manage.php:69]: Should this expect `activate` _or_ `deactivate` and throw if neither is present?

```php
if (isset($body['activate'])) {
    $this->setup->activateShop($body['activate']);
} elseif (isset($body['deactivate'])) {
    $this->setup->deactivateShop($body['deactivate']);
} else {
    throw ...
}
```

File: Setup/Uninstall.php [hunk 3]
```diff
@@ -66,7 +54,7 @@ public function uninstall(SchemaSetupInterface $setup, ModuleContextInterface $c
             $setup->startSetup();
             $metadatas = $this->config->getStoreMetadatas();
             foreach ($metadatas as $id => $metadata) {
-                $this->uninstallMerchant((int) $id, $metadata);
+                $this->setupHelper->disableShop((int) $id, $metadata);
```
Comment (rob-ns8) [inline Setup/Uninstall.php:57]: Should `disableShop` be `deactivateShop` here?

**Pull request merged**